test/test_tensor_creation_ops.py

```diff
@@ -339,6 +339,11 @@ def test_diagflat(self, device):
         expected = torch.eye(4, dtype=torch.complex128)
         self.assertEqual(result, expected)
 
+        # Complex support for torch.diag
```

Comment:
let's add test for `torch.diag` in a separate method

Reply:
Okay.
Any pointers which method?

Reply:
I think it's ok to add a new test method in this file `test_diag`

Reply:
Alright,  on it.

Reply:
Hey, @anjali411, I'm unable find the `test_diag`  file, is it the `test_shape_ops` ?

Reply:
Hey @Rajathbharadwaj you can just create a new method in the same file:
```
test_diag_complex(self, device):
     result = torch.diag(torch.ones(4, dtype=torch.complex128))
     expected = torch.eye(4, dtype=torch.complex128)
     self.assertEqual(result, expected)
```